### PR TITLE
fix: use backoff exception to gracefully exit and fail

### LINKED ISSUE #8219: Kokoro Infrastructure Failing on Periodic tests
The last few Kokoro periodic runs ([example](https://fusion.corp.google.com/runanalysis/buildlogs/prod:cloud-devrel%2Fpython-docs-samples%2Fpython3.7%2Fperiodic/prod:cloud-devrel%2Fpython-docs-samples%2Fpython3.7%2Fperiodic/KOKORO/c0ffa1af-4f56-4977-9175-73f41390b132/1658505438143/build%20%23949/Targets)) have ended without running  the tests:
```
=== Region tag injector setup complete ===
testing/btlr run --max-cmd-duration=30m **/requirements.txt -- /workspace/.kokoro/tests/run_single_test.sh
Collecting directories that match pattern...216 collected.
Running command(s)... [0 of 216 complete].

ERROR: Aborting VM command due to timeout of 18000 seconds
```

### PR DESCRIPTION
**Pull request by dandhlee**
## Description

The current retry is making tests fail out. Instead of indefinitely waiting, retrying with a specific number should help determine whether test passes / fails. 

Unfortunately I can't get the test to pass, but happy to incorporate ideas to help get this test passing. For now, they no longer will be timing out after ~5 hours, instead they'll fail or pass immediately.

Fixes #8219. Towards #7277.

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] Please **merge** this PR for me once it is approved.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

File: compute/oslogin/service_account_ssh_test.py [hunk 1]
```diff
@@ -84,19 +84,20 @@ def test_main(capsys):
         'oslogin', 'v1', cache_discovery=False, credentials=credentials)
     account = 'users/' + account_email
 
-    @retry(wait_exponential_multiplier=1000, wait_exponential_max=300000,
-           stop_max_attempt_number=10)
+    # Multiple exception types can be raised, using generic Exception as a catch all.
+    # I was not able to track all the exception types that can be raised from this block
+    # of code.
+    @backoff.on_exception(backoff.expo,
+                      Exception,
```
Comment (leahecole) [inline compute/oslogin/service_account_ssh_test.py:91]: could consider a tuple of multiple exceptions as an alternative to the generic

Comment (dandhlee) [reply compute/oslogin/service_account_ssh_test.py]: Done! Changed to include ones I could find for now, and additional ones could be easily added in the future.

**Review by engelke — APPROVED** [review]

**Comment by m-strzelczyk** [comment]
I think there is some curse on this issue, we're struggling with this for months now 🤣

**Comment by dandhlee** [comment]
It has been a tough one to crack indeed :(

**Review by m-strzelczyk — APPROVED** [review]
It won't get worse I guess ;)

**Comment by dandhlee** [comment]
Tests are expected to fail but admin merging to help tests get unblocked.

**Pull request merged**